src/prefect/deployments/runner.py

```diff
@@ -476,3 +612,188 @@ async def from_storage(
         cls._set_defaults_from_flow(deployment, flow)
 
         return deployment
+
+
+class DeploymentImage:
+    def __init__(self, name, tag=None, dockerfile="auto", build_kwargs=None):
+        self._name = name
+        self._tag = tag or slugify(pendulum.now("utc").isoformat())
```

Comment:
I would rather see all of these as public attributes given that they're exposed as `__init__` fields - allows users to sanity check like
```python
image = DeploymentImage(name="foo")
image.name # "foo"
```